metrics/metrics.go

```diff
@@ -80,36 +78,34 @@ func Init(cfg Config) (err error) {
 
 	switch cfg.Target {
 	case "stdout":
-		log.Print("Sending metrics to stdout")
 		return initStdout(cfg.Interval.Duration)
 	case "graphite":
 		if cfg.Addr == "" {
 			return errors.New("metrics: graphite addr missing")
 		}
-		log.Printf("Sending metrics to Graphite on %s as %q", cfg.Addr, pfx)
 		percentiles := cfg.Percentiles
 		if len(percentiles) == 0 {
 			percentiles = append(percentiles, []float64{0.75, 0.95, 0.99, 0.999}...)
 		}
 		return initGraphite(cfg.Addr, cfg.Interval.Duration, percentiles)
 	case "expvar":
 		handler := exp.ExpHandler(metrics.DefaultRegistry)
-		go startExpvar(cfg, handler)
-		return nil
+		return startExpvar(cfg, handler)
 
 	case "":
-		log.Printf("Metrics disabled")
 		return nil
 	default:
 		return fmt.Errorf("Invalid metrics target %s", cfg.Target)
 	}
 }
 
-func startExpvar(cfg Config, handler http.Handler) {
-	err := http.ListenAndServe(cfg.ExpAddr, handler)
-	if err != nil {
-		log.Printf("Could not start exp server: %q", err.Error())
-	}
+func startExpvar(cfg Config, handler http.Handler) error {
+	started := make(chan error)
+	go func() {
```

Comment:
Why do we need this gorutine? [`ListenAndServe`](https://golang.org/pkg/net/http/#ListenAndServe) is blocking and "always returns a non-nil error".